test/integration_test/types/testcase.go

```diff
@@ -18,17 +21,81 @@ type TestCase struct {
 	// started/failed to start correctly, and then remove it.
 	ShouldStartSuccessfully bool
 	// ShouldSkip check env requirements for the test case
-	ShouldSkip func() bool
+	ShouldSkip func(tc *TestCase) bool
 	// PureBackendRequirements contains additional information to construct a px-pure-secret.
 	PureBackendRequirements *PureBackendRequirements
 
 	// TODO: Add TestSetup and TestTearDown phase func when we have more cases
 }
 
+// TestReporter prints test results at the end of test runs
+// TODO: integration with Testrail in the future
+type TestReporter struct {
+	PassCases []TestCase
+	SkipCases []TestCase
+	FailCases []TestCase
+}
+
 // RunTest executes the actual test function
 func (tc *TestCase) RunTest(t *testing.T) {
 	if tc.ShouldSkip == nil {
-		tc.ShouldSkip = func() bool { return false }
+		tc.ShouldSkip = func(tc *TestCase) bool { return false }
+	}
+	if tc.ShouldSkip(tc) {
+		TestReporterInstance().AddSkipCase(*tc)
+		fmt.Printf("--- SKIP: %s %s\n", tc.TestName, tc.TestrailCaseIDs)
+		return
+	}
+
+	// NOTE: Tests filtered out by name regex will be marked as passed as well
+	passed := t.Run(tc.TestName, tc.TestFunc(tc))
+	if passed {
+		TestReporterInstance().AddPassCase(*tc)
+	} else {
+		TestReporterInstance().AddFailCase(*tc)
+	}
+}
+
+// TestReporterInstance returns the test reporter instance
+func TestReporterInstance() *TestReporter {
+	if testReporter == nil {
+		testReporter = &TestReporter{}
+	}
+	return testReporter
+}
+
+// AddPassCase adds passed test case to test reporter
+func (tp *TestReporter) AddPassCase(tc TestCase) {
+	TestReporterInstance().PassCases = append(TestReporterInstance().PassCases, tc)
```

Comment:
nit: We could just simplify this to -
```
tp.PassCases = append(tp.PassCases, tc)
```
Same for below, else there is no point of having the receiver.